src/include/s390_crypto.h

```diff
@@ -625,7 +625,7 @@ static inline int s390_kdsa(unsigned long func, void *param,
 	register unsigned long r1 asm("1") = (unsigned long)param;
 	register unsigned long r2 asm("2") = (unsigned long)src;
 	register unsigned long r3 asm("3") = (unsigned long)srclen;
-	unsigned long rc = 1;
+	unsigned long rc = 5;
```

Comment:
Why not use EIO here (instead of hardcoded 5 ) ????

Reply:
Because it does not compile ... errno.h is not yet included ... include it for this single place?

Reply:
... Or add a comment, like rc = 5; /* EIO */

Reply:
I just realized, that libica-api does not follow the best-practice for return values (rc/errno combination) like it is done in e.g. glibc. We should think about a change here for the next major release.

Reply:
I would prefer to leave the rc = 1 unchanged and do the maooing in the calling function. e.g.
```
if (s390_kdsa(...)) {
    rc == EIO;
    ...
}
```

Reply:
maybe `rc = s390_kdsa(....) ? EIO : 0;`  to keep the changes as small as possible. There are several places where kdsa is called.

Reply:
When looking at this, we have 7 routines where s390_kdsa is called. In 6 of them the rc from s390_kdsa is immediately changed/set to a desired value, based on rc == 0 or != 0. So no change necessary here.
The remaining place is in fact ecdsa_sign_cpacf (several occurrences in this routine). Here I would do a rc = s390_kdsa(....) ? EIO : 0; as Ingo suggested. 
Let me know if you agree.

Reply:
In `ecdsa_verify_cpacf()` we have also a conditional calling pattern, only the `rc` is changed to EFAULT. I agree to change it according to your proposal, but we should also agree upon the error value (EIO or EFAULT). In my opinion, EFAULT is not the right one (neither here, nor in `ecdsa_verify_cpacf()`), so I would vote for EIO.

BTW: Isn't changing the return value an API/ABI break?

Reply:
The verify case is different. Here `EFAULT` means that the signature is invalid (kind of an expected error case). 
`ecdsa_verify_cpacf()` is called by `ecdsa_verify_hw()` which is called by external API function `ica_ecdsa_verify()` and there `EFAULT` is documented as `signature invalid`. So this is part of the external API and can not be changed.

Reply:
from errno-base.h:
```
...
#define EIO              5      /* I/O error */
...
#define EFAULT          14      /* Bad address */
```

Reply:
Maybe `EINVAL`?

Reply:
EINVAL is already used for unsupported curve.

Reply:
With the recent push I fixed the rc = s390_kdsa(....) ? EIO : 0; for sign. No change for verify.